lib/vrl/stdlib/benches/benches.rs

```diff
@@ -1330,6 +1331,152 @@ bench_function! {
     }
 }
 
+bench_function! {
+    parse_xml => vrl_stdlib::ParseXml;
+
+    simple_text {
+        args: func_args![ value: r#"<a>test</a>"# ],
+        want: Ok(value!({ "a": "test" }))
```

Comment:
Coming from the JS world (and TSX, which is basically converting the dom to json), I'm more used to something like
`{ type: 'a', props: {}, children: 'test'}`
`{ type: 'a', props: {href: '...'}, children: ['text', {type: 'i', children: 'something'}]}`
It's probably more verbose though 🤔

Reply:
I should probably have mentioned in the PR, but one of the primary motivations for this feature is parsing XML for Datadog log lines per https://docs.datadoghq.com/logs/log_configuration/parsing/?tab=matchers#parsing-xml

The use-case this function enables is more general and doesn't _exactly_ match that format (for example, bools/numbers/null aren't strings by default), but can be configured that way with a few switches.

Another purpose is to make it somewhat queryable - e.g. `some.nested.key` instead of gymnastics to pull from the first variable of an array.